### PR TITLE
Playwright: Incorporate physical retry mechanism to GutenbergEditorPage.publish.

### LINKED ISSUE #57844: Playwright: apply the retry mechanism to posts to `GutenbergEditorPage.publish`.
### Details

Occasionally (like in TeamCity ID 6954527) there are issues where the editor states the post is published, but when visiting the published post link the `Oops! That page can't be found.` error is shown.

The bug is outlined in https://github.com/Automattic/wp-calypso/issues/57503.

In an attempt to correct for this inconsistency of the WPCOM/Calypso backend in the e2e test, apply the retry mechanism to this step.

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR inserts a check to verify that page content has loaded (instead of user being shown an error message like in https://github.com/Automattic/wp-calypso/issues/57503) and if not, use the physical retry mechanism to try a few times.

Key changes:
- rename the method `_visitPublishedEntryFromPublishPane` a shorter, much more legible `visitPublishedPost`.
- use `private` keyword for `visitPublishedPost`.
- move the `networkidle` wait into the `waitForNavigation` call.
- call the closure `confirmPostShown` after the initial navigation is complete.

#### Testing instructions

- [x] normal
  - [x] mobile
  - [x] desktop
  - [x] pre-release

Closes #57844.

### CODE REVIEW

**Comment by worldomonation** [comment]
The real test is to ensure that no more errors like https://github.com/Automattic/wp-calypso/issues/57503 show up.

File: packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts [hunk 1]
```diff
@@ -348,11 +349,31 @@ export class GutenbergEditorPage {
...
+		 * in the post not being visible to the public yet. In such cases, an error message is
+		 * instead shown to the user.
+		 *
+		 * When used in conjunction with `reloadAndRetry` this method will reload the page
+		 * multiple times to ensure the post content is shown.
+		 *
+		 * @param page
+		 */
+		async function confirmPostShown( page: Page ): Promise< void > {
+			await page.waitForSelector( '.entry-content' );
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts]: Using the default 30 seconds here seems pretty long before failing.... Maybe we could use the 5 second timeout we've used in other places?

**Review by dpasque — APPROVED** [review]
Overall change looks great! We should change the timeout on the wait, but once that is done, I think this is GTG! 🚀

File: packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts (hunk 1 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts]: Ah, good catch. Thanks!

Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts]: Fixed in d095ad112b86cc2c419a1bfd5b708d4a7366e917.

**Pull request merged**